### ISSUE TITLE
(matches) create view for initializing a match (Chloe and Nat)

### ISSUE THREAD

**Issue by elizabethli31** [uchicago-cs/chigame#27]
**Goal:** We currently lack a way for users to start matches once inside a lobby. We want a "Create New Match" view that allows users to initialize a match in a lobby with a minimum number of players or more. This view should have a field for the host user to choose which players in the lobby are going to participate in the match. This means the "create" view should automatically fill in the game attribute of the match (this will already be decided by the game associated with the lobby).

**Motivation:** Users should be able to decide when to start a match once inside a lobby.

**Open questions:**
1. Should only specific types of lobbies allow hosts to choose specific players for a match (while general lobbies include all users in the lobby as players by default)?
2. Should users be allowed to choose unregistered users as players?

**Comment by nugerhard** [comment]
This is the view that exists when you are looking to start a game. The game does not start on submission, rather the lobby which is prior to the match. This is essentially the "C" in CRUD as it relates to a game lobby.

**Comment by jiyeongchloe** [comment]
I've created a branch for this issue (`matches/match-initialization`).

So far, I can think of these steps to take:
- write a path to this create view in `urls.py`
    - URL can look something like `'lobby/<int:pk>/match/create/'`
- make a view class for `match_initialization` in `views.py`
    - we need to make sure we don't give the user access to `game`, `lobby`, and `winner` attributes of the form
- create a form class for `MatchInitializationForm` in `forms.py`
- create an HTML file in `templates/games/` directory

**Comment by jiyeongchloe** [comment]
committed changes to `matches/match-initialization` branch
- URL path in `urls.py`
- view definition for `match_init` in `views.py`
    - **this is only the default request definition. We need to implement code for restricting user access to `game`, `lobby`, and `winner` attributes**
- form class for `MatchInitForm` in `forms.py`
- HTML file `match_form.html`

**Comment by nugerhard** [comment]
Reviewing the `matches/match-initialization` branch, I think this match (as a data entity) creation task happens a lot farther down the line in the process than we initially thought. Lobby creation, aggregation, and detailed views of such still need to happen first. Maybe it's worth delaying this task to task 2?

Otherwise, the code looks good.

**Comment by jiyeongchloe** [comment]
Yeah, you're right about the other things needing to happen first. I think it's too late to delay the task to Sprint 2, so we can keep it as it is for this task. In any case, our original task was to just make a Create view, so we've done that. We can figure out how to incorporate it with the other parts in a later issue.

**Comment by jiyeongchloe** [comment]
created a pull request (PR #73) and assigned you @nugerhard as one of the reviewers

**Comment by jiyeongchloe** [comment]
based on comments on PR #73: 

I see what you're saying. I think users will need to create matches at some point; will elaborate on this in a comment in Issue #64 since it's more concerned with that issue (but the gist of it being: users should be given a choice to start a game once the lobby has the minimum number of players required. Otherwise, a match would start every time a minimum number of players is in the lobby, in which case we can't ever have 3, 4, or 5  players for a 2-5 player game (users might also want to play multiple matches in a lobby?))

**Comment by jiyeongchloe** [comment]
In-person meeting:
- decided on automatic creation of Match object instead of letting users tweak data

**issue is no longer relevant as we won't need a view for the user to see**
I am closing the issue, but will reopen if we decide to revisit

**Comment by elizabethli31** [comment]
Issue Score: Excellent

Comments:
Regardless of this code not being merged into dev, good work! Even though you guys decided to close the issue, you should still leave a closing statement. Also, the comments on issues should be a space for updates only. If you need to discuss how you plan on splitting the work or your ideas for the issue, that should be done on slack.
